incompatibility/apiset.go

```diff
@@ -44,13 +42,66 @@ func AggregateAnalysis(analysis ...*ApiSetIncompatibility) *ApiSetIncompatibilit
 	aggAnalysis := NewAnalysis()
 	for _, analysisObj := range analysis {
 		aggAnalysis.OpenApiFiles += analysisObj.OpenApiFiles
+		aggAnalysis.IncompatibleFiles += analysisObj.IncompatibleFiles
+		aggAnalysis.AnalysisPerIncompatibility =
+			mergeIncompatibilityAnalysis(aggAnalysis.AnalysisPerIncompatibility, analysisObj.AnalysisPerIncompatibility)
 	}
 	return aggAnalysis
 }
 
-// AggregateIncompatiblities aggrates information from individual incompatibilities into an
-// existing analysis object
-func AggregateIncompatibilities(analysis *ApiSetIncompatibility, incompatibilities ...*Incompatibility) *ApiSetIncompatibility {
+// FormAnalysis creates an analysis object from a single IncompatibilityReport
+func FormAnalysis(report *IncompatibilityReport, uniqueFilePath string) *ApiSetIncompatibility {
+	analysis := NewAnalysis()
 	analysis.OpenApiFiles++
+	intermedReport := CountIncompatibilities(report.GetIncompatibilities()...)
+	for class, count := range intermedReport.CountByClassification {
+		if count == 0 {
+			continue
+		}
+		fileOccurMap := analysis.AnalysisPerIncompatibility[class].CountPerFile
+		fileOccurMap[uniqueFilePath] =
+			&FileIncompatibilityClassificationAnalysis{
+				NumOccurrences: count}
+	}
+	failIncompatibilitiesCount := intermedReport.CountBySeverity[Severity_FAIL]
+	if failIncompatibilitiesCount > 0 {
+		analysis.IncompatibleFiles++
+	}
 	return analysis
 }
+
+// merge incompatibily analysis by classification
+func mergeIncompatibilityAnalysis(a1, a2 []*IncompatibilityAnalysis) []*IncompatibilityAnalysis {
+	var a3 []*IncompatibilityAnalysis = make([]*IncompatibilityAnalysis, len(IncompatibiltiyClassification_name))
+	for _, class := range IncompatibiltiyClassification_value {
+		mapCount1 := a1[class].CountPerFile
+		mapCount2 := a2[class].CountPerFile
+		a3[class] = &IncompatibilityAnalysis{
+			IncompatibilityClass: IncompatibiltiyClassification(class),
+			CountPerFile:         appendFileInfomation(mapCount1, mapCount2),
+		}
+	}
+	return a3
+}
+
+// adds incompatibility counts by file in m2 to m1 each map should correspond to the same
+// incompatibility classification
+func appendFileInfomation(m1, m2 map[string]*FileIncompatibilityClassificationAnalysis) map[string]*FileIncompatibilityClassificationAnalysis {
+	for fileName, incompAnalysis := range m2 {
+		m1[fileName] = incompAnalysis
+	}
+	return m1
+}
+
+func CountIncompatibilities(incompatibilities ...*Incompatibility) IntermediateReport {
```

Comment:
I think this can be private. The tests can use this method because they are in the same package.

Reply:
used in TestFileInformationInclude in analysis package,
here's my thinking for including it in the analysis testing file
- analysis should test that the process from reports to one analysis object is correct